server/mailrepository/mailrepository-api/src/main/java/org/apache/james/mailrepository/api/MailRepositoryFactory.java

```diff
@@ -0,0 +1,29 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailrepository.api;
+
+import org.apache.james.mailrepository.api.MailRepository;
+import org.apache.james.mailrepository.api.MailRepositoryUrl;
+
+public interface MailRepositoryFactory {
+    Class<? extends MailRepository> fqdn();
```

Comment:
```suggestion
    Class<? extends MailRepository> clazz();
```
Or...

```suggestion
    Class<? extends MailRepository> mailRepositoryClass();
```

?

Feels strange we never used fqdn in the code base just yet.

Reply:
I guess it should have been `fqcn` to match the `fullyQualifiedClassName` used in[ GuiceMailRepositoryLoader](https://github.com/apache/james-project/blob/master/server/container/guice/common/src/main/java/org/apache/james/modules/server/GuiceMailRepositoryLoader.java#L41). 
I'll change it to `mailRepositoryClass`

Reply:
fixed up